sim/midas/src/main/scala/midas/passes/WriteXDCFile.scala

```diff
@@ -0,0 +1,102 @@
+// See LICENSE for license details.
+
+package midas.passes
+
+import firrtl._
+import firrtl.stage.Forms
+import firrtl.annotations._
+import firrtl.analyses.InstanceKeyGraph
+
+import midas.stage.{GoldenGateFileEmission}
+import midas.targetutils.xdc._
+
+/**
+  * We could reuse [[GoldenGateOutputFileAnnotation]] here, but this makes it
+  * marginally easier to filter out.
+  */
+private[midas] case class XDCOutputAnnotation(fileBody: String, suffix: Option[String])
+    extends NoTargetAnnotation with GoldenGateFileEmission {
+  def getBytes = fileBody.getBytes
+}
+
+private[midas] object WriteXDCFile extends Transform with DependencyAPIMigration with XDCAnnotationConstants {
+    override def prerequisites = Forms.LowForm
+  // Probably should run before emitter?
```

Comment:
Since you handle this in https://github.com/firesim/firesim/pull/825/files#diff-a402c9c5cbea2ce9c9b948d9fe3ad00021373ce9ac1313f597cb4a8b63d71bc5R43-R46, I'd vote for removing this comment.
```suggestion
```